### PR TITLE
duplicate json-api tests for full-lifecycle user token testing

### LINKED ISSUE #12907: whether the many users of the  test utility function continue to work if that function returns a user token

### PR DESCRIPTION
**Pull request by realvictorprm**
Shall fix #12907

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json/src/failure/scala/http/FailureTests.scala [hunk 1]
```diff
@@ -96,40 +96,48 @@ final class FailureTests
...
-      (status, output) <- postJsonStringRequestEncoded(
-        uri.withPath(Uri.Path("/v1/create")),
-        body.compactPrint,
-        headersWithParties(List(p.unwrap)),
+      (status, output) <- headersWithParties(List(p.unwrap)).flatMap(
+        postJsonStringRequestEncoded(
+          uri.withPath(Uri.Path("/v1/create")),
+          body.compactPrint,
+          _,
+        )
```
Comment (S11001001) [inline ledger-service/http-json/src/failure/scala/http/FailureTests.scala:118]: This is an alternative that might work, but only if you think it is a more obvious and reviewable refactoring.

```suggestion
      (status, output) <- postJsonStringRequestEncoded(
          uri.withPath(Uri.Path("/v1/create")),
          body.compactPrint,
          _,
        ) =<< headersWithParties(List(p.unwrap))
```

Comment (realvictorprm) [reply ledger-service/http-json/src/failure/scala/http/FailureTests.scala:118]: The flatMap operator might do the job here too, at least to get rid of all the braces.

Comment (realvictorprm) [reply ledger-service/http-json/src/failure/scala/http/FailureTests.scala]: decided to not go with operators, it didn't make much of difference :(

Comment (S11001001) [reply ledger-service/http-json/src/failure/scala/http/FailureTests.scala]: I was really thinking about the diff, if headersWithParties is closer to where it was it might be easier to read. But, again, whatever works out best in practice 🤷

File: ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala [hunk 2]
```diff
@@ -183,8 +188,10 @@ trait AbstractHttpServiceIntegrationTestFuns
 
   protected def withHttpService[A](
       f: (Uri, DomainJsonEncoder, DomainJsonDecoder, LedgerId) => Future[A]
-  ): Future[A] =
+  ): Future[A] = {
+    println("======> inside withHttpService")
```
Comment (S11001001) [inline ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala:192]: Leftover println, best leave a `TODO` here for now.

File: ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala [hunk 3]
```diff
@@ -862,9 +1055,9 @@ abstract class AbstractHttpServiceIntegrationTest
       jsObject(
         """{"templateIds": ["Iou:Iou", "UnknownModule:UnknownEntity"], "query": {"currency": "EUR"}}"""
       )
-
-    search(List(), query, uri, encoder, headersWithPartyAuth(List("UnknownParty"))).map {
-      response =>
+    logger.info("query returns unknown Template IDs")
```
Comment (S11001001) [inline ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala:1058]: What's this for?

File: ledger-service/http-json/src/itlib/scala/http/AbstractWebsocketServiceIntegrationTest.scala [hunk 4]
```diff
@@ -21,12 +28,12 @@ import scalaz.syntax.tag._
 import scalaz.syntax.traverse._
 import scalaz.{-\/, \/-}
 import spray.json.{
+  DeserializationException,
   JsArray,
   JsNull,
   JsObject,
   JsString,
   JsValue,
-  DeserializationException,
```
Comment (S11001001) [inline ledger-service/http-json/src/itlib/scala/http/AbstractWebsocketServiceIntegrationTest.scala:29]: Intellij spam, should revert.

File: ledger-service/http-json/src/itlib/scala/http/AbstractWebsocketServiceIntegrationTest.scala [hunk 5]
```diff
@@ -43,12 +50,12 @@ abstract class AbstractWebsocketServiceIntegrationTest
     with AbstractHttpServiceIntegrationTestFuns
     with BeforeAndAfterAll {
 
-  import HttpServiceTestFixture._
+//  import HttpServiceTestFixture._
```
Comment (S11001001) [inline ledger-service/http-json/src/itlib/scala/http/AbstractWebsocketServiceIntegrationTest.scala:53]: TODO here

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/endpoints/RouteSetup.scala [hunk 6]
```diff
@@ -210,18 +202,15 @@ private[http] object RouteSetup {
     ).run(fn)
 
   private[http] def handleFutureFailure[A](fa: Future[A])(implicit
-      ec: ExecutionContext,
-      lc: LoggingContextOf[InstanceUUID with RequestID],
+      ec: ExecutionContext
   ): Future[Error \/ A] =
-    fa.map(a => \/-(a)).recover(logException("Future") andThen Error.fromThrowable andThen (-\/(_)))
+    fa.map { a => \/-(a) }.recover(Error.fromThrowable andThen (-\/(_)))
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/endpoints/RouteSetup.scala:207]: Some of this appears to be from another PR...?

**Review by S11001001 — COMMENTED** [review]
Very nice, I especially appreciate the test code cleanups (default arg removal &c).

**Comment by realvictorprm** [comment]
Let's try getting something in and continue further work in a follow-up PR.

File: ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala [hunk 7]
```diff
@@ -678,8 +766,76 @@ trait AbstractHttpServiceIntegrationTestFuns
...
+      admin: Boolean = false,
+  )(implicit ec: ExecutionContext): Future[Jwt] =
+    Future.successful(
+      HttpServiceTestFixture.jwtForParties(actAs, readAs, ledgerId, withoutNamespace)
+    )
+
+  protected def jwt(uri: Uri)(implicit ec: ExecutionContext): Future[Jwt] =
+    jwtForParties(uri)(List("Alice"), List(), testId)
+
+  protected val jwtAdminNoParty: Jwt = {
```
Comment (S11001001) [inline ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala:791]:
```suggestion
  protected lazy val jwtAdminNoParty: Jwt = {
```

File: ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala (hunk 3 above)
Comment (S11001001) [reply ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala:1007]: f2d8fcdf76f

**Review by S11001001 — APPROVED** [review]

File: ledger-service/http-json-oracle/src/it/scala/http/HttpServiceWithOracleIntTest.scala [hunk 8]
```diff
@@ -17,3 +17,7 @@ class HttpServiceWithOracleIntTest
 
   override def wsConfig: Option[WebsocketConfig] = None
 }
+
+final class HttpServiceWithOracleIntTestCustomToken
+    extends HttpServiceWithOracleIntTest
+    with AbstractHttpServiceIntegrationTestFunsCustomToken
```
Comment (ray-roestenburg-da) [inline ledger-service/http-json-oracle/src/it/scala/http/HttpServiceWithOracleIntTest.scala:23]: Maybe rename `AbstractHttpServiceIntegrationTestFunsCustomToken` to `AbstractHttpServiceIntegrationTestCustomToken`
